skimage/filters/_multiotsu.pyx

```diff
@@ -0,0 +1,54 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+import numpy as np
+
+cimport numpy as cnp
+cimport cython
+from itertools import combinations
+
+def _find_threshold_multiotsu(double [:, ::1] var_btwcls,
+                              Py_ssize_t classes,
+                              Py_ssize_t bins,
+                              double [::1] aux_thresh):
```

Comment:
In your function, you are doing alot of allocations inside cython. I suggest you allocate `aux_thresh` in cython too!

Reply:
@hmaarrfk I actually tend to prefer all allocations to happen in Cython. That makes it clearer where gil is needed vs not needed, and whether your Cython is optimal (since you should have exactly 0 yellow lines inside the function).

Reply:
Sure. Do the allocations how you want, but don't Pass and return the same thing.